Add 2022-11-17 minutes (#80)
solid/notifications-panel: meetings/2022-11-17.md

* SC: We'll allocate some time for implementation feedback. Links to servers/applications and demos welcome.
* eP: webhooks should land in CSS 6.0 https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1388#issuecomment-1318295827
* SC: We should discuss WH during next meeting maybe invite Jackson

ACTION: eP to reach out to Jackson

### Release of the spec
URL: https://github.com/solid/notifications-panel/blob/main/meetings/2022-11-10.md#release-of-the-spec

>ACTION: SC to review https://hackmd.io/3sLhQvmKTVKQKoXeXLSTpA
>ACTION: SC to create a project board for the releases.

* SC: Still a todo.

### Update Notifications UCR
URL: https://github.com/solid/notifications-panel/issues/76

>SC: I'll review https://github.com/solid/notifications-panel/pull/79

* SC: In the middle of reviewing.
* SC: Checking to see if one is covered by existing. Not sure yet.
* RG: There are multiple design choices. Making UCs explicit constraints the design space.
* eP: Seems like reasonable UCs we can iterate on.

ACTION : eP to review.

* eP: Approved.
* SC: https://github.com/solid/notifications-panel/issues/76 will focus on editorial and other iterations.

> * eP: Some of them may be about managing connections. That may be too much for the use case but use connections for the same client or subscriber, or for multiple connections. Especially for websockets the server may want to reuse same connection for the same client. This sounds like a requirement for me no matter how exactly we solve it.

Reviewer: ```suggestion
* eP: Some of them may be about managing connections. That may be too much for the use case but reuse connections for the same client or subscriber, or for multiple connections. Especially for websockets the server may want to reuse same connection for the same client. This sounds like a requirement for me, no matter how exactly we solve it.
```

* eP: Which parts of these go into the UC or too much detail for the UC.
* RG: More like requirement than a UC document based on real-world stuff. We need the practical UCs and then the requirements out of that. Based on that we need to justify some design decisions.

### Simplify flow diagrams
URL: https://github.com/solid/notifications/pull/108

* SC: Any objections to merge?
* eP: It is still approved from me and has requested changes from Sarven
* SC: I see it is the loop part. No objections to merge it with it in.
* SC: Rahul are you ok with this?
* RG: +1 I'm good with it.
* SC: Merging

### Considerations

* SC: We may be able to process these security considerations quickly.
* RG: You need to explain what is the context here.

#### Add consider-exposing-information
URL: https://github.com/solid/notifications/pull/117

> * SC: Exposing information advises to avoid exposing any information that is not required for the interaction. For example when notification is provided, it may include which resource was changed but not which agent changed it. We don't want servers to just slip in such information.

Reviewer: ```suggestion
* SC: Exposing information advises to avoid exposing any information that is not required for the interaction. For example, when notification is provided, it may include which resource was changed but not which agent changed it. We don't want servers to just slip in such information.
```

* RG: Do you plan to include it as normative? I think that for the server it should maybe be required.
* SC: We don't want to say that something is non-conforming just because something added some information.
* ...: When someone is writing implementation they can use it as a reference when they review it.

> * ...: In some environments some constraints may be provided, and something would needed to be included in the payload. We don't want to prohibit such use scenarios.

Reviewer: ```suggestion
* ...: In some environments some constraints may be provided, and something would need to be included in the payload. We don't want to prohibit such use scenarios.
```

* eP: Makes sense to me. I also don't think we can make it normative.
* RG: I need to think about it and see if we need to add something normative for the server.
* SC: We don't want to limit in what ways the protocol can be used. We could mandate closed list of information but that would be very limiting.

> * RG: For the client I'm not worrying about what it sends. For the server I see three buckets, what must be send and whay may be send. Te former we can specify what must not be send.

Reviewer: ```suggestion
* RG: For the client I'm not worrying about what it sends. For the server I see three buckets: what must be sent; what may be sent; what must not be sent.
```

> * SC: I don't think we can specify what must not be send.

Reviewer: ```suggestion
* SC: I don't think we can specify what must not be sent.
```

* eP: Do you have some concrete example?
* RG: For example exposing access control.
* SC: It is possible that administrator or owner may want to track changes to access control.

> * RG: In the document I created I made a list of what would be the minimum information that has to be send. We could take a look at it.

Reviewer: ```suggestion
* RG: In the document I created, I made a list of what would be the minimum information that has to be sent. We could take a look at it.
```

#### Add consider-subscription-resource-trust
URL: https://github.com/solid/notifications/pull/118

> * SC: Client should be careful where it makes request. Especially a localhost, if I know that you have a server you your localhost, I could made you make request to your local server repeatedly.

Reviewer: ```suggestion
* SC: Client should be careful where it makes requests. Especially a localhost. if I know that you have a server on your localhost, I could make you repeatedly make requests to your local server.
```

* RG: How do you stop this on the web. I can point to subscription on your server and clients would go make subscriptions there. Now that server would start getting subscription requests.

> * SC: There are legitimate reasons to have something on localhost. We just want to prevent clients blindly making requests. For example that client can have allow and deny lists of origins or particular resources. 

Reviewer: ```suggestion
* SC: There are legitimate reasons to have something on localhost. We just want to prevent clients blindly making requests. For example, that client can have allow and deny lists of origins or particular resources. 
```

* eP: I think we need separate considerations for clients to be restrained with their retries.
* ...: I don't see clear threat with localhost so some reference there would be ok.

> * SC: This is mostly about making requests where you trust. In HTTP there is some consideration for redirects. Browser follows redirects automatically but for example curl needs a flag.

Reviewer: ```suggestion
* SC: This is mostly about making requests where you trust. In HTTP, there is some consideration for redirects. Browser follows redirects automatically, but, for example, `curl` needs a flag.
```

* ...: I can think of some parental lock scenarios where warning might be needed.

> * eP: For me talking about trust here is too much, I don't see how implementations should put this consideration in pratctice.

Reviewer: ```suggestion
* eP: For me, talking about trust here is too much. I don't see how implementations should put this consideration in practice.
```

* SC: I recall Webmention recommends not sending them to localhost. https://www.w3.org/TR/webmention/#avoid-sending-webmentions-to-localhost

> * RG: I want to come back to CORS. In a manner similar to redirects the Access-Control-Allow-Origin header explicitly advises clients which domains it is allow to fetch data from. Perhaps a similar scheme is needed here?

Reviewer: ```suggestion
* RG: I want to come back to CORS. In a manner similar to redirects, the `Access-Control-Allow-Origin` header explicitly advises clients which domains it is allowed to fetch data from. Perhaps a similar scheme is needed here?
```

> * SC: Different case I think. Client may not trust the subscription resource. Consider client running into a resource and the user wants to subscribe to updates e.g., updates from an intended political party, but then ends up subscribing to updates from an unwanted political party. This advisory is a warning so that client can factor in whatever they need to determine whether to trust the subscription resource or not before they send the subscription request.

Reviewer: ```suggestion
* SC: Different case I think. Client may not trust the subscription resource. Consider client running into a resource and the user wants to subscribe to updates, e.g., from an intended political party, but ends up subscribing to updates from an unwanted political party. This advisory is a warning so that client can factor in whatever they need to determine whether or not to trust the subscription resource before they send the subscription request.
```
